src/Nancy/DefaultObjectSerializer.cs

```diff
@@ -17,18 +34,33 @@ public string Serialize(object sourceObject)
         {
             if (sourceObject == null)
             {
-                return String.Empty;
+                return string.Empty;
             }
 
-            var formatter = new BinaryFormatter();
+            dynamic serializedObject = null;
 
-            using (var outputStream = new MemoryStream())
+            //In the case of adding values to cookies it is a string else an object for sessions

```

Comment:
I think what we should do here is

1. Kill the comment
2. Break out the `else`-block into a method
3. Change it to

```
serializedObject = (sourceObject is string)
   ? sourceObject
   : this.WhatEverTheMethodIsCalled(....)
```